src/main/scala/org.constellation.cmhotwallet/CoMakerClient.scala

```diff
@@ -0,0 +1,135 @@
+package org.constellation.cmhotwallet
+
+import java.security.SecureRandom
+import java.time.Instant
+
+import cats.data.EitherT
+import org.http4s.circe.CirceEntityEncoder._
+import org.http4s.circe.CirceEntityDecoder._
+import io.circe.syntax._
+import cats.effect.{ConcurrentEffect, Resource, Sync}
+import cats.implicits._
+import org.constellation.cmhotwallet.model.{CMKeyPair, Transfer, Transaction => CMTransaction}
+import org.constellation.cmhotwallet.model.config.{CliConfig, CoMakeryConfig}
+import org.constellation.cmhotwallet.Ed25519KeyTool.publicKeyToBase64
+import org.constellation.cmhotwallet.model.request._
+import org.http4s.Method.{GET, POST, PUT}
+import org.http4s.Uri.{Authority, RegName, Scheme}
+import org.http4s.client.Client
+import org.http4s.{Header, Headers, Request, Uri}
+import org.http4s.util.CaseInsensitiveString
+
+class CoMakerClient(keyTool: Ed25519KeyTool) {
+  import CoMakerClient._
+
+  private val secureRandom: SecureRandom = try {
+    SecureRandom.getInstance("NativePRNGNonBlocking")
+  } catch {
+    case _: Throwable => SecureRandom.getInstanceStrong
+  }
+
+  private def nonce: String = BigInt(120, secureRandom).toString(16)
+
+  private def timestamp: String = Instant.now.getEpochSecond.toString
+
+  private def getUri(path: String)(config: CoMakeryConfig): Uri =
+    Uri(scheme = Some(Scheme.http), authority = Some(Authority(host = RegName(config.host), port = config.port.some)))
+      .addPath(config.apiSuffix)
+      .addPath(path)
```

Comment:
Can you make it in more functional way? The rest of code is written in such manner so it would be good to keep it unified over the codebase.

Reply:
It looks like a builder, but actually underneath it uses .copy, so it's functional in the end.